src/Pos/Client/Txp/Addresses.hs

```diff
@@ -0,0 +1,14 @@
+{-# LANGUAGE TypeFamilies #-}
+
+module Pos.Client.Txp.Addresses
+       ( MonadAddresses (..)
+       ) where
+
+import           Universum
+
+import           Pos.Types (Address)
+
+-- | A class which have the method to generate a new address
+class Monad m => MonadAddresses m where
+    type AddrData m :: *
+    getNewAddress :: AddrData m -> m Address
```

Comment:
I really don't like this. I'd rather have `getNewAddress :: m Address`, but the only way for implementing such instance for real wallet which I could think of is having some `newtype UglyWrapper m = UglyWrapper (ReaderT addrData m) deriving (<thousands of monads>)` and implementing an instance for it. Though we used to do such things, now it's a no-way, as I understand. Is there a better way to implement such a typeclass?